server/command_hooks_test.go

```diff
@@ -424,3 +425,98 @@ func TestExecuteCommand(t *testing.T) {
 		wg.Wait()
 	})
 }
+
+type MockUser struct {
+	mock.Mock
+}
+
+func (m *MockUser) HasPermissionToChannel(userID, channelID string, permission *model.Permission) bool {
+	args := m.Called(userID, channelID, permission)
+	return args.Bool(0)
+}
+
+func (m *MockUser) HasPermissionTo(userID string, permission *model.Permission) bool {
+	args := m.Called(userID, permission)
+	return args.Bool(0)
+}
+
+func (m *MockUser) Get(userID string) (*model.User, error) {
+	args := m.Called(userID)
+	return args.Get(0).(*model.User), args.Error(1)
+}
+
+func TestHasPermissionToExportChannel(t *testing.T) {
+	mockUser := new(MockUser)
+	clientWrapper := &pluginapi.Wrapper{
+		User: mockUser,
+	}
+	plugin := &Plugin{
+		client: clientWrapper,
+	}
+
+	tests := []struct {
+		name                    string
+		enableAdminRestrictions bool
+		userID                  string
+		channelID               string
+		hasPermissionToChannel  bool
+		hasPermissionToSystem   bool
+		expected                bool
+	}{
+		{
+			name:                    "Admin restrictions enabled, has permission to channel",
+			enableAdminRestrictions: true,
+			userID:                  "user1",
+			channelID:               "channel1",
+			hasPermissionToChannel:  true,
+			hasPermissionToSystem:   false,
+			expected:                true,
+		},
+		{
+			name:                    "Admin restrictions enabled, has permission to system",
+			enableAdminRestrictions: true,
+			userID:                  "user2",
+			channelID:               "channel2",
+			hasPermissionToChannel:  false,
+			hasPermissionToSystem:   true,
+			expected:                true,
+		},
+		{
+			name:                    "Admin restrictions enabled, lacks both permissions",
+			enableAdminRestrictions: true,
+			userID:                  "user3",
+			channelID:               "channel3",
+			hasPermissionToChannel:  false,
+			hasPermissionToSystem:   false,
+			expected:                false,
+		},
+		{
+			name:                    "Admin restrictions disabled",
+			enableAdminRestrictions: false,
+			userID:                  "user4",
+			channelID:               "channel4",
+			hasPermissionToChannel:  false,
+			hasPermissionToSystem:   false,
+			expected:                true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.enableAdminRestrictions {
+				mockUser.On("HasPermissionToChannel", tt.userID, tt.channelID, model.PermissionManageChannelRoles).Return(tt.hasPermissionToChannel)
+				if !tt.hasPermissionToChannel {
+					mockUser.On("HasPermissionTo", tt.userID, model.PermissionManageSystem).Return(tt.hasPermissionToSystem)
+				}
```

Comment:
Does it need to be nested?

Reply:
Yes, if `hasPermissionToChannel` is true then only the call to `HasPermissionTo` is made.